rust/syntect-plugin/src/main.rs

```diff
@@ -273,54 +296,70 @@ impl<'a> PluginState<'a> {
     }
 }
 
-impl<'a> state_cache::Plugin for PluginState<'a> {
-    type State = State;
 
-    fn initialize(&mut self, ctx: PluginCtx<State>, _buf_size: usize) {
-        self.do_highlighting(ctx);
+impl<'a> Plugin for Syntect<'a> {
+    type Cache = StateCache<LineState>;
+
+    fn new_view(&mut self, view: &mut View<Self::Cache>) {
+        let _t = trace_block("Syntect::new_view", &["syntect"]);
+        eprintln!("added view {:?}", view.get_id());
```

Comment:
there's random logging in the original, and this seems very relevant, but at some point we probably want to make this more systematic